lib/utils/export/to-zip.ts

```diff
@@ -81,18 +81,17 @@ const appendTags = (note) => {
  * // when given `Yummy Recipe` and `Yummy Recipe` as duplicates
  * // will return `Yummy Recipe` and `Yummy Recipe (1)`
  *
- * @param {[Array, Object]} accumulator list of note objects for export and filename counts
+ * @param {[Array, Map]} accumulator list of note objects for export and filename counts
  * @param {Object} note note object
  * @returns {[Array, Object]} final note list and accumulating filename counts
  */
 const toUniqueNames = ([notes, nameCounts], note) => {
-  const newNameCounts = update(nameCounts, note.fileName, (n) =>
-    n || 0 === n ? n + 1 : 0
-  );
-  const count = newNameCounts[note.fileName];
+  nameCounts.set(note.fileName, (nameCounts.get(note.fileName) ?? -1) + 1);
```

Comment:
why start at -1? is it to avoid creating a file with `-0`?

could we reverse the order of these queries?

```ts
count = nameCounts.get(note.fileName) ?? 0;
noteCounts.set(note.fileName, count + 1);
```

Reply:
Yup, that is why I used the -1. Is there a reason you prefer not to use the -1? Is it for readability or something else I'm missing?

Reply:
`-1` stands out to me because it's a value we know we'll never want. there is no occasion where we'll want a negative number appended to a filename. appearing in the source code then stands out as a mismatch; it breaks the symmetry between the concept and implementation.

put another way, it's a clue that maybe we overlooked something about how we were doing it. in this case I think the discrepancy was related to the ordering of our data dependencies and something as simple as swapping those lines resulted in an interface that doesn't suffer from the question that led us to introduce `-1` and it eliminates the need to read _twice_ from the map to get the count.

because of these things it does seem more readable to me, because it doesn't prompt me to ask, "now why is that there?"

Reply:
Thank you!